src/test/java/se/simonsoft/cms/indexing/properties/HandlerSubjectSchemeTest.java

```diff
@@ -0,0 +1,130 @@
+import se.simonsoft.cms.item.properties.CmsItemPropertiesMap;
+import se.simonsoft.cms.xmlsource.handler.s9api.XmlSourceReaderS9api;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class HandlerSubjectSchemeTest {
+
+	private final Processor processor = new Processor(false);
+	private final XmlSourceReaderS9api sourceReader = new XmlSourceReaderS9api(processor);
+    private final HandlerSubjectScheme handler = new HandlerSubjectScheme(sourceReader, processor);
+	private String xml = 	"<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" +
+							"<?xml-model href=\"urn:oasis:names:tc:dita:spec:classification:rng:subjectScheme.rng\" schematypens=\"http://relaxng.org/ns/structure/1.0\"?>\n" +
+							"<subjectScheme>\n" +
+							"    <hasInstance>\n" +
+							"        <subjectdef keys=\"productname\">\n" +
+							"            <topicmeta>\n" +
+							"                <shortdesc>Product Name</shortdesc>\n" +
+							"            </topicmeta>\n" +
+							"            <subjectdef keys=\"Tempo_T_6\">\n" +
+							"                <topicmeta>\n" +
+							"                    <navtitle>Tempo T 6</navtitle>\n" +
+							"                </topicmeta>\n" +
+							"            </subjectdef>\n" +
+							"            <subjectdef keys=\"Rapid_A_600J\"/>\n" +
+							"            <subjectdef keys=\"Inspire_1200S\">\n" +
+							"                <topicmeta>\n" +
+							"                    <navtitle>Inspire 1200S</navtitle>\n" +
+							"                </topicmeta>\n" +
+							"            </subjectdef>\n" +
+							"        </subjectdef>\n" +
+							"    </hasInstance>\n" +
+							"    \n" +
+							"    <enumerationdef>\n" +
+							"        <elementdef name=\"cms:property\"/>\n" +
+							"        <attributedef name=\"cds:productname\"/>\n" +
+							"        <subjectdef keyref=\"productname\"/>\n" +
+							"    </enumerationdef>\n" +
+							"</subjectScheme>";
+
+	@Test
+	public void testHandlerProjectLevel() {
+		IndexingDoc doc = new IndexingDocIncrementalSolrj();
+		CmsChangesetItem item = mock(CmsChangesetItem.class);
+		when(item.isFile()).thenReturn(true);
+		when(item.getPath()).thenReturn(new CmsItemPath("/dita/xml/bookmap1.ditamap"));
+		IndexingItemProgress progress = mock(IndexingItemProgress.class);
+		CmsItemPropertiesMap properties = new CmsItemPropertiesMap();
+		when(progress.getProperties()).thenReturn(properties);
+		when(progress.getFields()).thenReturn(doc);
+		CmsRepository repo1 = new CmsRepository("http://cmshostname/svn/keydefmap1");
+		when(progress.getRepository()).thenReturn(repo1);
+		when(progress.getItem()).thenReturn(item);
+		properties.and("cds:productname", "Inspire_1200S Tempo_T_6 Rapid_A_600J");
```

Comment:
Add test coverage of a key that is unknown to the subject scheme.

Reply:
Done.